Sources/HaCWebsiteLib/Utils/GitUtil.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import Dispatch
+import DotEnv
+
+/// A utility to manage read-only interactions with external Git repos
+public struct GitUtil {
+  /// Used to avoid multiple git pulls being run simultaneously
+  private let serialQueue: DispatchQueue
+
+  /// The directory that will hold all the repos we clone
+  private let cloneRoot = DotEnv.get("DATA_DIR")!
+
+  private let remoteRepoURL: String
+  private let directoryName: String
+
+  public var localRepoPath: String {
+    return cloneRoot + "/" + directoryName
+  }
+
+  init(remoteRepoURL: String, directoryName: String) {
+    serialQueue = DispatchQueue(label: "com.hac.website.\(directoryName)-pull-queue")
+    self.remoteRepoURL = remoteRepoURL
+    self.directoryName = directoryName
+  }
+
+  /// A safe version of `unsafeUpdate`
+  public func update(then callback: @escaping () throws -> Void = {}) rethrows {
```

Comment:
It this meant to be asynchronous? Because it doesn't appear to behave that way. What are the goals of this PR?

Reply:
Good catch! This is intended to by synchronous and I think this makes sense since users of this API can always build asynchronous behaviour on top of that when they need it. I've fixed this up now.
The goals of this PR is to make it easy to pull many repos into disk and not worry about the dispatch stuff. The main focus was moving the dispatch queuing behaviour into the GitUtil rather than having 'WorkshopManager' deal with it directly. So now for a given repository, calling `update` a bunch of times will block until each `update` call has been able to execute sequentially.